stl/inc/execution

```diff
@@ -5110,29 +5110,30 @@ _NoThrowFwdIt uninitialized_default_construct_n(_ExPo&& _Exec, _NoThrowFwdIt _Fi
         _STD advance(_First, _Count);
         return _First;
     } else {
-        return _STD for_each_n(
-            _STD forward<_ExPo>(_Exec), _First, _Count, [](_Ty& _Obj) { _STD _Default_construct_in_place(_Obj); });
+        return _STD for_each_n(_STD forward<_ExPo>(_Exec), _First, _Count,
+            [](auto& _Obj) { _STD _Default_construct_in_place(reinterpret_cast<_Ty&>(_Obj)); });
     }
 }
 
 _EXPORT_STD template <class _ExPo, class _NoThrowFwdIt, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
 void uninitialized_value_construct(_ExPo&&, _NoThrowFwdIt _First, _NoThrowFwdIt _Last) noexcept /* terminates */ {
     // value-initialize all elements in [_First, _Last)
-    _REQUIRE_CPP17_MUTABLE_LVALUE_ITERATOR(_NoThrowFwdIt);
+    _REQUIRE_PARELLEL_LVALUE_ITERATOR(_NoThrowFwdIt);
     _STD _Adl_verify_range(_First, _Last);
     auto _UFirst      = _STD _Get_unwrapped(_First);
     const auto _ULast = _STD _Get_unwrapped(_Last);
 
-    using _UnwrappedNoThrowFwdIt = _Unwrapped_t<const _NoThrowFwdIt&>;
-    if constexpr (_Use_memset_value_construct_v<_UnwrappedNoThrowFwdIt>) {
+    if constexpr (_Use_memset_value_construct_v<decltype(_UFirst)>) {
         _STD _Zero_range(_UFirst, _ULast);
     } else if constexpr (remove_reference_t<_ExPo>::_Parallelize) {
         const size_t _Hw_threads = _CSTD __std_parallel_algorithms_hw_threads();
         if (_Hw_threads > 1) { // parallelize on multiprocessor machines...
             auto _Count = _STD distance(_UFirst, _ULast);
             if (_Count >= 2) { // ... with at least 2 elements
                 _TRY_BEGIN
-                auto _Ctor_fn = [](decltype(*_UFirst) _Obj) { _STD _Construct_in_place(_Obj); };
+                auto _Ctor_fn = [](decltype(*_UFirst) _Obj) {
+                    _STD _Construct_in_place(reinterpret_cast<_Iter_value_t<_NoThrowFwdIt>&>(_Obj));

```

Comment:
Lines 5150, 5155, 5159, 5198, 5204, and 5209 need to similarly `reinterpret_cast` the result of dereferencing. (Extracting `_Iter_value_t<_NoThrowFedIt>` is probably a good idea for readability.)